work_queue/src/work_queue.c

```diff
@@ -3432,6 +3445,34 @@ static char *expand_envnames(struct work_queue_worker *w, const char *payload)
 	return expanded_name;
 }
 
+/*
+Send a url or remote command used to generate a cached file,
+if it has not already been cached there.  Note that the length
+may be an estimate at this point and will be updated by return
+message once the object is actually loaded into the cache.
+*/
+
+static work_queue_result_code_t send_special_if_not_cached( struct work_queue *q, struct work_queue_worker *w, struct work_queue_task *t, struct work_queue_file *tf, const char *typestring )
+{
+	if(hash_table_lookup(w->current_files,tf->cached_name)) return WQ_SUCCESS;
+
+	char source_encoded[WORK_QUEUE_LINE_MAX];
+	char cached_name_encoded[WORK_QUEUE_LINE_MAX];
+
+	url_encode(tf->payload,source_encoded,sizeof(source_encoded));
+	url_encode(tf->cached_name,cached_name_encoded,sizeof(cached_name_encoded));
+
+	send_worker_msg(q,w,"%s %s %s %d %o\n",typestring, source_encoded, cached_name_encoded, tf->length, 0777);
```

Comment:
Should we add input/output flag for future use of output files?

Reply:
Changing these messages to `inurl` and `incmd` to correspond to `infile` and anticipating future `outurl` and `outcmd`

Reply:
Correction: `puturl` and `putcmd` to correspond to `put` in the cache management protocol.  `infile` and `outfile` are in the task management specification, and solely concern the link between cache directory and sandbox.